app/controllers/customers_controller.rb

```diff
@@ -59,18 +60,29 @@ def update
     else
       render :edit, status: :unprocessable_entity
     end
+    # Not sure I need this, I don't know what I would update that would cause error.
+    rescue ActiveRecord::InvalidForeignKey => e
+      flash.alert = "Error updating customer: #{e.message}"
+      render :edit, status: :unprocessable_entity
+    end
   end
 
   # DELETE /customers/1 or /customers/1.json
   def destroy
-    @customer.destroy
-
-    respond_to do |format|
-      format.html { redirect_to customers_url, notice: "Customer was successfully destroyed." }
-      format.json { head :no_content }
+    begin
+      @customer.destroy
+      redirect_to customers_path, notice: 'Customer was successfully deleted.'
+    # Added so that a customer with orders cannot be deleted. Removed dependent: :destroy on model.
+    rescue ActiveRecord::InvalidForeignKey
+      redirect_to customers_path, alert: 'Unable to delete a customer with existing orders.'
     end
   end
 
+  def orders
+    @customer = Customer.find(params[:id])
+    @orders = @customer.orders
+  end
```

Comment:
I suppose this action was for this part of the assignment?

> Change the show view for the customer, so that if that customer has orders, a list of them are shown in a table at the bottom, with show/update/delete links.

If so, well, the assignment is asking for a change the existing show view, i.e., you should see the list of orders at <http://localhost:3000/customers/1> as opposed to <http://localhost:3000/customers/1/orders>. However, even though what you did is different, I think this is still fine because the goal of the assignment is to learn, and you got a little extra practice with the new route, action, and view.